lib/jobs/redfish-discovery.js

```diff
@@ -245,33 +246,67 @@ function RedfishDiscoveryJobFactory(
                 targetList.push(target);
             });
             
+            var identifiers = [];
             var config = Object.assign({}, self.settings);
             config.root = data.system['@odata.id'];
-            var node = {
-                type: 'compute',
-                name: data.system.Name,
-                identifiers: [ data.system.Id ],
-            };
             
-            var relations = [{
-                relationType: 'enclosedBy',
-                targets: targetList
-            }];
-
-            var obm = {
-                config: config,
-                service: 'redfish-obm-service'
-            };
-            
-            return self.upsertRelations(node, relations)
-            .then(function(nodeRecord) {
-                return Promise.all([
-                    waterline.obms.upsertByNode(nodeRecord.id, obm),
-                    nodeRecord
-                ]);
+            return self.redfish.clientRequest(config.root)
+            .then(function(res) {
+                var ethernet = _.get(res.body, 'EthernetInterfaces');
+                if(ethernet) {
```

Comment:
Yes, Redfish spec defines EthernetInterfaces as an object type,   it should never be an array.